itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/RunProjectWithDebugTest.java

```diff
@@ -0,0 +1,177 @@
+import org.eclipse.reddeer.swt.impl.tree.DefaultTreeItem;
+import org.eclipse.reddeer.workbench.impl.editor.TextEditor;
+import org.jboss.tools.quarkus.core.QuarkusCorePlugin;
+import org.jboss.tools.quarkus.integration.tests.project.universal.methods.AbstractQuarkusTestMethods;
+import org.jboss.tools.quarkus.reddeer.common.QuarkusLabels.TextLabels;
+import org.jboss.tools.quarkus.reddeer.perspective.QuarkusPerspective;
+import org.jboss.tools.quarkus.reddeer.ui.launch.QuarkusLaunchConfigurationTabGroup;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * 
+ * @author olkornii@redhat.com Oleksii Korniienko
+ * 
+ */
+@OpenPerspective(QuarkusPerspective.class)
+@RunWith(RedDeerSuite.class)
+public class RunProjectWithDebugTest extends AbstractQuarkusTestMethods {
+
+	private static String PROJECT_NAME = "testRunWithDebug";
+	private static String RESOURCE_PATH = "src/main/java";
+	private static String ORG_ACME = "org.acme";
+	private static String EXAMPLE_RESOURCE = "ExampleResource.java";
+	private static String VARIABLE = "i";
+	private static String FIRST_LINE = "    	int " + VARIABLE + " = 5;";
+	private static String SECOND_LINE = "    	" + VARIABLE + " = 7;";
+
+	@BeforeClass
+	public static void testNewNewQuarkusMavenProject() {
+		testCreateNewProject(PROJECT_NAME, TextLabels.MAVEN_TYPE);
+		checkJdkVersion(PROJECT_NAME, TextLabels.MAVEN_TYPE);
+
+		checkProblemsView();
+	}
+
+	@Test
+	public void testRunWithDebug() {
+
+		ProjectItem exampleResource = new ProjectExplorer().getProject(PROJECT_NAME).getProjectItem(RESOURCE_PATH)
+				.getProjectItem(ORG_ACME).getProjectItem(EXAMPLE_RESOURCE);
+
+		insertLines(exampleResource, FIRST_LINE, SECOND_LINE);
+		addBreakpointToLine(exampleResource, FIRST_LINE);
+		addBreakpointToLine(exampleResource, SECOND_LINE);
+
+		new QuarkusLaunchConfigurationTabGroup().selectProject(PROJECT_NAME);
+		new QuarkusLaunchConfigurationTabGroup().openDebugConfiguration();
+
+		new DefaultTreeItem(TextLabels.QUARKUS_APPLICATION_TREE_ITEM).select();
+		new ContextMenuItem("New Configuration").select();
+
+		new PushButton(TextLabels.DEBUG).click();
+
+		ConsoleView consoleView = new ConsoleView();
+		new WaitUntil(new ConsoleHasText(consoleView, "[io.quarkus]"), TimePeriod.getCustom(600));
+
+		URL localhost = null;
+		try {
+			localhost = new URL("http://localhost:8080/hello");
+		} catch (MalformedURLException e) {
+			QuarkusCorePlugin.logException("Wrong URL! ", e);
+		}
+
+		openDebugPerspective();
+		Thread thread = startProject(localhost);
+		try {
+			Thread.sleep(1000); // w8 for read localhost from child thread
+		} catch (InterruptedException e) {
+			QuarkusCorePlugin.logException("Thread interupted!", e);
+			Thread.currentThread().interrupt();
+		}
+
+		new ShellMenuItem("Run", "Step Over").select();
+		int firstValue = checkVariable(VARIABLE);
+		new ShellMenuItem("Run", "Step Over").select();
+		int secondValue = checkVariable(VARIABLE);
```

Comment:
I suggest to do this a little bit differently: you should try to change variable during debug, so debugger should stop at breakpoint, afterwards you should change variable in variables view, go to next breakpoint, check if the variable is changed in output(either use console output or content on URL).

Reply:
rdy